src/staverepetition.ts

```diff
@@ -121,24 +125,32 @@ export class Repetition extends StaveModifier {
     ctx.save();
     ctx.setFont(this.textFont);
 
-    // Default to right symbol
-    let text_x = 0 + this.x_shift;
-    let symbol_x = x + this.x_shift;
-    if (this.symbol_type === Repetition.type.CODA_LEFT) {
-      // Offset Coda text to right of stave beginning
-      text_x = this.x + stave.getVerticalBarWidth();
-      symbol_x = text_x + ctx.measureText(text).width + 12;
-    } else if (this.symbol_type === Repetition.type.DS) {
-      const modifierWidth = stave.getNoteStartX() - this.x;
-      text_x = this.x + x + this.x_shift + stave.getWidth() - 5 - modifierWidth - ctx.measureText(text).width;
-      // TODO this is weird. setting the x position should probably be refactored, parameters aren't clear here.
-    } else {
-      // Offset Signo text to left stave end
-      symbol_x = this.x + x + stave.getWidth() - 5 + this.x_shift;
-      text_x = symbol_x - +ctx.measureText(text).width - 12;
+    let text_x = 0;
+    let symbol_x = 0;
+    const modifierWidth = stave.getNoteStartX() - this.x;
+    switch (this.symbol_type) {
+      // To the left with symbol
+      case Repetition.type.CODA_LEFT:
+        // Offset Coda text to right of stave beginning
+        text_x = this.x + stave.getVerticalBarWidth();
+        symbol_x = text_x + ctx.measureText(text).width + 12;
```

Comment:
Can we move this hardcoded `12` to the metrics file please?

Reply:
done